ic-asset/src/asset_config.rs

```diff
@@ -214,6 +238,18 @@ impl AssetConfigTreeNode {
             .filter(|rule| rule.applies(canonical_path))
             .fold(base_config, |acc, x| acc.merge(x))
     }
+
+    fn is_included(&self, canonical_path: &Path) -> bool {
+        let included_in_parent = match &self.parent {
+            Some(parent) => parent.is_included(canonical_path),
+            None => false,
+        };
+        self.rules
+            .iter()
+            .filter(|rule| rule.applies(canonical_path))
+            .any(|rule| !rule.ignore_hidden)
+            || included_in_parent
+    }
```

Comment:
I think we can simplify this by removing the "hidden" filtering from the directory walk, and instead merging the `ignore` field into the AssetConfig and filtering using that.

Reply:
i'll explore this idea tomorrow

Reply:
done in https://github.com/dfinity/agent-rs/pull/361/commits/ffc5abe70058880f9f92c5ebfb3f215944e834ed

turned out great!